test/testswmr.pwsh.in

```diff
@@ -0,0 +1,535 @@
+# Copyright by The HDF Group.
+# Copyright by the Board of Trustees of the University of Illinois.
+# All rights reserved.
+#
+# This file is part of HDF5.  The full HDF5 copyright notice, including
+# terms governing use, modification, and redistribution, is contained in
+# the COPYING file, which can be found at the root of the source code
+# distribution tree, or in https://www.hdfgroup.org/licenses.
+# If you do not have access to either file, you may request a copy from
+# help@hdfgroup.org.
+#
+# Tests for the swmr feature.
+
+$srcdir  = '@srcdir@'
+$bindir  = '@bindir@'
+$testdir = '@testdir@'
+
+###############################################################################
+## test parameters
+###############################################################################
+
+$Nreaders     = 5          # number of readers to launch
+$Nrdrs_spa    = 3          # number of sparse readers to launch
+$Nrecords     = 200000     # number of records to write
+$Nrecs_rem    = 40000      # number of times to shrink
+$Nrecs_spa    = 20000      # number of records to write in the sparse test
+$Nsecs_add    = 5          # number of seconds per read interval
+$Nsecs_rem    = 3          # number of seconds per read interval
+$Nsecs_addrem = 8          # number of seconds per read interval
+$nerrors      = 0
+
+###############################################################################
+## definitions for message file to coordinate test runs
+###############################################################################
+$WRITER_MESSAGE = 'SWMR_WRITER_MESSAGE'    # The message file created by writer that the open is complete
+                                           # This should be the same as the define in "$bindir/swmr_common.h"
+$MESSAGE_TIMEOUT = 300                     # Message timeout length in secs
+                                           # This should be the same as the define in "./h5test.h"
+
+# Print a line-line message left justified in a field of 70 characters
+# beginning with the word "Testing".
+#
+function Test-WithSpaces {
+   $SPACES='                                                               '
+   #Write-Output  "Testing $* $SPACES" | cut -c1-70 | tr -d '\012'
+}
+
+# To wait for the writer message file or till the maximum # of seconds is reached
+# $Message is the message file to wait for
+# This performs similar function as the routine h5_wait_message() in test/h5test.c
+function Wait-Message {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory, ValueFromPipeline)]
+        [string]$Message                              # Get the name of the message file to wait for
+    )
+
+    PROCESS {
+        $message = Join-Path -Path $testdir -ChildPath $Message
```

Comment:
I think $testdir needs to passed in as well - $testdir gets changed later to the test folder.